src/cls/rbd/cls_rbd_types.h

```diff
@@ -133,6 +133,7 @@ inline void decode(MirrorImageStatusState &state, bufferlist::iterator& it)
 
 struct MirrorImageStatus {
   MirrorImageStatus() {}
+  ~MirrorImageStatus() {}
```

Comment:
>There are no virtual methods -- no need to specify virtual on the destructor.

Isn't that just a rule of thunb?

Meyers says in Effective C++, 2nd ed:
"The C++ language standard is unusually clear on this topic: when you try to delete a derived class object through a base class pointer and the base class has a nonvirtual destructor (as EnemyTarget does), the results are undefined. This means compilers may generate code to do whatever they like: reformat your disk, send suggestive mail to your boss, fax source code to your competitors, whatever. (What often happens at runtime is that the derived class's destrutcor is never called. ...)"

Sorry if I'm wrong.